src/Microsoft.Extensions.Caching.Memory/MemoryCacheServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for setting up memory cache related services in an <see cref="IServiceCollection" />.
+    /// </summary>
+    public static class MemoryCacheServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds memory caching services to the specified <see cref="IServiceCollection" />.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        public static void AddMemoryCache(this IServiceCollection services)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            services.AddOptions();
+            services.TryAdd(ServiceDescriptor.Transient<IDistributedCache, MemoryDistributedCache>());
```

Comment:
First, what @Tratcher said. Second, just because we happen to have certain components with certain requirements doesn't make them quite such broad concepts. In-memory caching and distributing caching are two completely different features. In some sense they shouldn't even be in the same abstractions package, but meh.